Tasks/VsTest/VSTest.ps1

```diff
@@ -41,23 +41,19 @@ $codeCoverage = Convert-String $codeCoverageEnabled Boolean
 
 if($testAssemblyFiles)
 {
-    Write-Verbose "Calling Invoke-VSTest for all test assemblies"
-    $timeline = Start-Timeline -Context $distributedTaskContext
-    $projectName = Get-Variable -Context $distributedTaskContext -Name "System.TeamProject"
-    $buildDir = Get-Variable -Context $distributedTaskContext -Name "Agent.BuildDirectory" -Global $FALSE
-    $buildNumber = Get-Variable -Context $distributedTaskContext -Name "Build.BuildId"
-    $buildUri = Get-Variable -Context $distributedTaskContext -Name "Build.BuildUri"
-    $owner = Get-Variable -Context $distributedTaskContext -Name "Build.RequestedFor"	
-    $cwd = $buildDir
-    $testResultsDir = $buildDir+"\"+"TestResults"
-    Write-Verbose "Calling Invoke-VSTest from working folder: $cwd"
-    Invoke-VSTest -TestAssemblies $testAssemblyFiles -Timeline $timeline -VSTestVersion $vsTestVersion -TestFiltercriteria $testFiltercriteria -RunSettingsFile $runSettingsFile -PathtoCustomTestAdapters $pathtoCustomTestAdapters -CodeCoverageEnabled $codeCoverage -OverrideTestrunParameters $overrideTestrunParameters -OtherConsoleOptions $otherConsoleOptions -WorkingFolder $cwd -TestResultsFolder $testResultsDir
-    $connection = Get-VssConnection -TaskContext $distributedTaskContext
-    if($connection)
-    {
-        $resultFiles = Find-Files -SearchPattern "*.trx" -RootFolder $testResultsDir
-        Invoke-ResultPublisher -Connection $connection -ProjectName $projectName -Owner $owner -ResultFiles $resultFiles -ResultType "Trx" -BuildUri $buildUri -BuildNumber $buildNumber -Platform $platform -Configuration $configuration
-    }
+    Write-Verbose -Verbose "Calling Invoke-VSTest for all test assemblies"
+
+    $artifactsDirectory = Get-Variable -Context $distributedTaskContext -Name "System.ArtifactsDirectory" -Global $FALSE
```

Comment:
There is a variable called System.TestResultsDirectory set in the build extension that I think you would want to use.  The advantage is it will automatically get cleaned up before the build starts each time.